skyllh/core/pdf.py

```diff
@@ -675,26 +691,31 @@ def __init__(self, axis_binnings, pdf_path_to_splinetable=None, pdf_grid_data=No
                 vmax=axis_binning.upper_edge
             ))
 
-       
-        pdf_loaded = False
+        # need either splinetable or value grid
+        if((pdf_path_to_splinetable is None) and (pdf_grid_data is None)):
+            raise TypeError(
+                "missing input. need at least one of the following: path to splinetable or grid data!")
+
+        # if available prefer to work with photosplines otherwise use gridinterpolator
+        if(isinstance(pdf_path_to_splinetable, str) and PHOTOSPLINE_LOADED):
+            self._pdf = photospline.SplineTable(pdf_path_to_splinetable)
 
-        # if available prefer to work with photosplines
-        if(isinstance(pdf_path_to_splinetable, str)):
-            if(PHOTOSPLINE_LOADED):
-                self._pdf = photospline.SplineTable(pdf_path_to_splinetable)
-                pdf_loaded = True
-                self._photospline = True
+        else:
+            if(isinstance(pdf_path_to_splinetable, str) and not PHOTOSPLINE_LOADED):
+                raise ImportError(
+                    "missing library: can not find photosplines.")
 
+            if not isinstance(pdf_grid_data, np.ndarray):
+                raise TypeError("missing or wrong input: can not create RegularGridInterpolator from type {}".format(
```

Comment:
add expected type to error message

Reply:
resolved in https://github.com/IceCubeOpenSource/skyllh/pull/22/commits/d33951c1cff6af86a1771b651c729295daee7fde